### PR TITLE
pythonPackages.mask-rcnn: init at 2.1

### PR DESCRIPTION
**Pull request by Rakesh4G**
###### Motivation for this change
pythonPackages.mask-rcnn: init at 2.1

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
@CMCDragonkai

### CODE REVIEW

**Comment by Rakesh4G** [comment]
currently dependency tensor-flow is failing.

File: pkgs/development/python-modules/mask-rcnn/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    rev = "3deaec5d902d16e1daf56b62d5971d428dc920bc";
+    sha256 = "13s3q9yh2q9m9vyksd269mww3bni4q2w7q5l419q70ca075qp8zp";
+  };
+      
+  propagatedBuildInputs = [
+    imgaug
+    numpy
+    scipy
+    pillow
+    cython
```
Comment (jonringer) [inline pkgs/development/python-modules/mask-rcnn/default.nix]: cython should be in `nativeBuildInputs`

File: pkgs/development/python-modules/mask-rcnn/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    matplotlib
+    scikitimage
+    Keras
+    opencv3
+    h5py
+    ipython
+    tensorflow
```
Comment (jonringer) [inline pkgs/development/python-modules/mask-rcnn/default.nix]: not related to PR, but this is like a list of "most likely to be broken" packages.... xD.

Tensorflow especially lately.

File: pkgs/development/python-modules/mask-rcnn/default.nix (hunk 1 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/mask-rcnn/default.nix]: done.

**Comment by jonringer** [comment]
unfortunately, I don't see tensorflow getting fixed anytime soon

**Comment by CMCDragonkai** [comment]
I suspect that `mask-rcnn` can be used with cuda or without cuda. This means that we should have a cuda version as well. This is similar to the other packages that have `WithCuda` as a suffix. However because the name of this is `mask-rcnn` then I guess it's more like `mask-rcnn-with-cuda` and `mask-rcnn`.

Also this should be kept as a library, not application.

Then `mask-rcnn-with-cuda` will have `tensorflow` substituted with `tensorflowWithCuda`.

The only other package that has `WithCuda` that isn't the tensorflow/theano/pytorch is `colmap` where there's `colmapWithCuda`. It could make sense that this only applies to applications, and libraries should be as the basic version.

**Comment by Rakesh4G** [comment]
Hi @jonringer , 

This is now fixed.
The broken tensorflow dependency has been fixed.
https://github.com/NixOS/nixpkgs/pull/81633

@CMCDragonkai

**Comment by FRidh** [comment]
We typically avoid top-level variations of a package when options are already present.

File: pkgs/development/python-modules/mask-rcnn/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    opencv3
+    pillow
+    scikitimage
+    scipy
+    tensorflow
+  ];
+
+  meta = with lib; {
+    description = "Mask R-CNN for object detection and instance segmentation on Keras and TensorFlow";
+    homepage = https://github.com/matterport/Mask_RCNN;
```
Comment (jonringer) [inline pkgs/development/python-modules/mask-rcnn/default.nix]:
```suggestion
    homepage = "https://github.com/matterport/Mask_RCNN";
```

**Review by jonringer — COMMENTED** [review]
otherwise diff LGTM,

still waiting on tensorflow to not be broken to ensure the package works

File: pkgs/development/python-modules/mask-rcnn/default.nix (hunk 3 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/mask-rcnn/default.nix]: ok

**Review by jonringer — APPROVED** [review]
diff LGTM
commit LGTM

```
[4 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/80520
1 package marked as broken and skipped:
python38Packages.mask-rcnn

1 package built:
python37Packages.mask-rcnn
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python37Packages.mask-rcnn
